### PR TITLE
Use docker-compose to introduce Redis

### PR DESCRIPTION
**Pull request by eckdanny-osi**
See https://osi-cwds.atlassian.net/browse/CASE-1015

~HELP from a rails person plz...~

_UPDATE_: 

Adds a `docker-compose` configuration with default networking to allow i/o between our `app` service and the `redis` service. Adds some basic rails-* redis gems (👀 rails peeps plz evaluate).

### CODE REVIEW

**Comment by eckdanny-osi** [comment]
As of a4297f5 (just pushed) I can verify:

 - the `compose` config exposes the redis service to rails
 - the the Rails config @mlipardo and I elected to use doesn't cause an exception

How I verified the docker-compose set up:

1) start up the composed environment:

 ```sh
docker-compose build
docker-compose start
```

2) add something to redis store from rails container

```sh 
docker-compose exec web rails console

irb(main)> my_redis = Redis.net(:host => 'redis')
=> #<Redis client v4.0.1 for redis://redis:6379/0>
irb(main)> my_redis.set "foo", [1, 2, 3].to_json
=> "OK"
```

3) retrieve the same something from store in the redis container

```
docker-compose exec redis redis-cli

127.0.0.1:6379> get foo
"[1,2,3]"
```

**Comment by mlipardo** [comment]
@eckdanny-osi I think there is a typo `my_redis = Redis.new(:host => 'redis')`

**Comment by eckdanny-osi** [comment]
@mlipardo in the [comment thread](https://github.com/ca-cwds/case-management/pull/111#issuecomment-352547595)? Yes. Updated ~`Redis.net(...`~ -> `Redis.new(...`)

File: Gemfile [hunk 1]
```diff
@@ -38,6 +38,12 @@ gem 'jbuilder', '~> 2.5'
 # Use Capistrano for deployment
 # gem 'capistrano-rails', group: :development
 #
+# Use redis for stuff
```
Comment (edgibbs) [inline Gemfile]: remove comment

File: config/environments/development.rb [hunk 2]
```diff
@@ -16,7 +16,11 @@
   if Rails.root.join('tmp/caching-dev.txt').exist?
     config.action_controller.perform_caching = true
 
-    config.cache_store = :memory_store
+    # Danny and Melbourne don't know what we're doing
```
Comment (edgibbs) [inline config/environments/development.rb]: Let's remove this since we don't have a need for a cache in redis yet.

**Comment by eckdanny-osi** [comment]
@edgibbs removed all rails redis config in 36ead57bce6610a1c3cb238c016711bd25352141

**Comment by eckdanny-osi** [comment]
TIL what the _dismiss review_ button does. Was worried might lose line-item feedback but its still [here](https://github.com/ca-cwds/case-management/pull/111#pullrequestreview-84533732).

**Pull request merged**